File: src/BuilderPattern/CarBuilder/CarDirector.java
package BuilderPattern.CarBuilder;

import java.awt.Color;

// Car Director build premade Car-configurations
public class CarDirector {

    // Default-Values of the CarBuilder (optional)
    private static final String defaultBrand = "";
    private static final String defaultModel = "";
    private static final int defaultAmountOfDoors = 4;
    private static final Color defaultColor = Color.BLACK;

    // Method to build the default car (optional)
    public ICarBuilder buildDefaultCar(ICarBuilder builder) {
        return builder.brand(defaultBrand)
                .model(defaultModel)
                .amountOfDoors(defaultAmountOfDoors)
                .color(defaultColor);
    }

    // Method to build BMW's
    public void buildBMW(ICarBuilder builder) {
        buildDefaultCar(builder).brand("BMW");
    }

    // Method to build Blue Audi's
    public void buildBlueAudi(ICarBuilder builder) {
        buildDefaultCar(builder).brand("Audi").color(Color.BLUE);
    }

    // Build White Smart's
    public void buildWhiteSmart(ICarBuilder builder) {
        buildDefaultCar(builder).brand("Smart").color(Color.WHITE).amountOfDoors(2);
    }

}

File: BuilderPattern/CSharp/NSCarBuilder/CarDirector.cs
using System.Drawing;

namespace BuilderPattern.NSCarBuilder {

    // Car Director build premade Car-configurations
    internal class CarDirector {

        // Default-Values of the CarBuilder (optional)
        private static string defaultBrand = "";
        private static string defaultModel = "";
        private static int defaultAmountOfDoors = 4;
        private static Color defaultColor = Color.Black;

        // Method to build the default car (optional)
        public ICarBuilder BuildDefaultCar(ICarBuilder builder) {
            return builder.Brand(defaultBrand)
                .Model(defaultModel)
                .AmountOfDoors(defaultAmountOfDoors)
                .Color(defaultColor);
        }

        // Method to build BMW's
        public void BuildBMW(ICarBuilder builder) {
            BuildDefaultCar(builder).Brand("BMW");
        }

        // Method to build Blue Audi's
        public void BuildBlueAudi(ICarBuilder builder) {
            BuildDefaultCar(builder).Brand("Audi").Color(Color.Blue);
        }

        // Build White Smart's
        public void BuildWhiteSmart(ICarBuilder builder) {
            BuildDefaultCar(builder).Brand("Smart").Color(Color.White).AmountOfDoors(2);
        }
    }
}
